app/controllers/user.js

```diff
@@ -16,12 +16,38 @@ const passwordValid = password => {
   return isAlphanumeric.test(password) && password.length >= 8;
 };
 
+const generateUser = (user, next) => {
+  if (!emailValid(user.email)) {
+    logger.error(`Email: ${user.email} invalid.`);
+    return next(errors.emailNotValid(user.email));
+  }
+  if (!passwordValid(user.password)) {
+    logger.error('Password invalid.');
+    return next(errors.passwordInvalid);
+  }
+  logger.info(`All validations passed, going to create the user: ${JSON.stringify(user)}`);
+  return bcrypt.hash(user.password, saltRounds).then(hash => {
+    user.password = hash;
+    return User.createModel(user);
+  });
+};
+const giveAdminPriviledges = user => {
+  return User.update({ admin: true }, { returning: true, where: { email: user.email } });
+};
+
 exports.adminSignUp = (req, res, next) => {
+  const user = {
```

Comment:
remember that this isn't a user, its some user attributes or params, but its not a user. It's a little difference but it will avoid misunderstandings if you name things in the most accurate way possible.